### PR TITLE
[RDY] Travis: also upload coverage to codecov

### PR DESCRIPTION
**Pull request by blueyed**
codecov supports flags and provides better reports.

I think it makes sense to use it.

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [run]
 source = pinax
-omit = pinax/stripe/conf.py,pinax/stripe/tests/*,pinax/stripe/migrations/*
+omit = pinax/stripe/conf.py,pinax/stripe/migrations/*
```
Comment (paltman) [inline .coveragerc]: I’d like to exclude tests from coverage.

Comment (blueyed) [reply .coveragerc]: Isn't it good to see if they are actually 100% covered, i.e. executed?

**Comment by paltman** [comment]
No. I just care about coverage of the app code. We know tests will run.

**Comment by blueyed** [comment]
> No. I just care about coverage of the app code. We know tests will run.

But it can show issues with the tests, e.g. https://coveralls.io/builds/13442770/source?filename=pinax%2Fstripe%2Ftests%2Ftest_managers.py#L111.

**Comment by blueyed** [comment]
> The travis_retry command

Why?
Otherwise the `--required` might make the build fail if there is a network issue.
It is good practice to also wrap e.g. `pip install` with it.
It is a function provided by Travis.

**Comment by paltman** [comment]
Hmmm. This is a very good point. You changed my mind.

**Comment by paltman** [comment]
Pinax is moving everything over to CircleCI

**Comment by blueyed** [comment]
So should I remove the other omits as well?

> pinax/stripe/conf.py,pinax/stripe/migrations/*

There is some uncovered code in there, but I think it should be rather marked there (`coverage: omit` - why not use the defaults, i.e. `# pragma: no cover`) - or get covered after all in the long run.

**Comment by paltman** [comment]
there is a lot of different infrastructure decisions being made in here that I'm not sure now is the right time.  i'm glad to fast track adding codecov support, but not while it's bundled with other things.  

1. not sure why the naming change in the travis.yml file but we'll be moving to circleci soon so I don't really care about the travis changes.  just find them unnecessary at this point.
2. if we want to start monitoring things we've previously omitted, and i see your argument for doing so, then i don't want to merge until the PR also resolves them so I'd like to see a separate PR that removes the omits and includes the commits necessary to make them covered.

**Comment by blueyed** [comment]
Ok, adjusted.

**Pull request merged**